build_tools/azure/test_script.sh

```diff
@@ -57,6 +57,10 @@ if [[ "$SHOW_SHORT_SUMMARY" == "true" ]]; then
     TEST_CMD="$TEST_CMD -ra"
 fi
 
+if [[ "$CHECK_MEMORY" == "true" ]]; then
+    TEST_CMD="$TEST_CMD --min-ram=50"
```

Comment:
To avoid long and verbose CI reports I would use a higher limit by default:

```suggestion
    TEST_CMD="$TEST_CMD --min-ram=250"
```

If we need to get an exhaustive memory usage report, it's possible to do a local run.

I think it's fine to have tests that allocate up to 250 MB.